internal/racetime/api.go

```diff
@@ -69,19 +79,114 @@ type CategoryResponse struct {
+		TimesRaced   string `json:"times_raced"`
+	}
+}
+
+type UserSearchParameters struct {
+	Name string
+	Scrim string
+	Term string
+}
+
 // CategoryDetail fetches the race data of a specific racetime.gg category
 func CategoryDetail(c config.Racetime, category string) (*CategoryResponse, error) {
-	resp, err := http.Get(fmt.Sprintf("%s/%s/data", c.URL, category))
+	content, err := Get(c, fmt.Sprintf("%s/data", category), nil)
+
+	var cr *CategoryResponse
+	json.Unmarshal([]byte(content), cr)
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
 
-	var cr *CategoryResponse
-	err = json.NewDecoder(resp.Body).Decode(cr)
+	return cr, nil
+}
+
+func CategoryLeaderboards(c config.Racetime, category string) (*LeaderboardsResponse, error) {
+	content, err := Get(c, fmt.Sprintf("%s/leaderboards/data", category), nil)
+
+	var cl *LeaderboardsResponse
+	json.Unmarshal([]byte(content), cl)
 	if err != nil {
 		return nil, err
 	}
 
-	return cr, nil
+	return cl, nil
 }
+
+// RaceDetail fetches the race data of a specific racetime.gg race
+func RaceDetail(c config.Racetime, category string, race string) (*RaceData, error) {
+	content, err := Get(c, fmt.Sprintf("%s/%s/data", category, race), nil)
+
+	var rd *RaceData
+	json.Unmarshal([]byte(content), rd)
+	if err != nil {
+		return nil, err
+	}
+
+	return rd, nil
+}
+
+func PastUserRaces(c config.Racetime, user string, showEntrants bool, page int){
+	content, err := Get(c, fmt.Sprintf("user/%s/races/data", user), nil)
+
+	var pur *RaceData
+	json.Unmarshal([]byte(content), pur)
+	if err != nil {
+	}
+}
+
+func UserSearch(c config.Racetime, user string){
+	parameters := UserSearchParameters{
+		Name: "colfra",
+	}
+
+	m := structs.Map(parameters)
+	content, _ := Get(c, "user/search", m)
+	var userSearchResults UserDataResponse
+	json.Unmarshal([]byte(content), &userSearchResults)
+	log.Println(userSearchResults.Results[0].ID)
+}
+
+func GetTest(){
+	var test = `{"results": [{"id": "5BRGVMd30E368Lzv", "full_name": "colfra", "name": "colfra", "discriminator": null, "url": "/user/5BRGVMd30E368Lzv", "avatar": null, "pronouns": null, "flair": "staff", "twitch_name": null, "twitch_display_name": null, "twitch_channel": null, "can_moderate": false}]}`;
+	jd := json.NewDecoder(strings.NewReader(test))
+	var userResults UserDataResponse
+	jd.Decode(&userResults)
+	log.Println(userResults.Results[0].ID)
+}
+
+func Get(c config.Racetime, endpoint string, parameters map[string]interface{}) (string, error) {
+	requestUrl, err := url.Parse(fmt.Sprintf("%s/%s", c.URL, endpoint))
+	if (len(parameters) != 0){
+		q := requestUrl.Query()
+		for key, value := range parameters {
+			if (value != ""){
+				q.Add(strings.ToLower(key), value.(string))
+			}
+		}
+		requestUrl.RawQuery = q.Encode()
+	}
+
+	resp, err := http.Get(requestUrl.String())
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+    if err != nil {
+        log.Fatal(err)
```

Comment:
As a general rule I'd say only ever `log.Fatal` from `func main()`. It's better to just bubble this error up via a return than to crash your app and hope you have a recovery layer somewhere.